### DISCUSSION TITLE
Wsl/Service/CreateInstance/ERROR_FILE_NOT_FOUND

### DISCUSSION THREAD

**Discussion by frankli-dev** [microsoft/WSL#8729] [General]
### Version

Microsoft Windows [Version 10.0.22622.450]

### WSL Version

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

Ubuntu

### Distro Version

Ubuntu

### Other Software

None

### Repro Steps

1. Install wsl default distro
2. Uninstall wsl
3. Reinstall wsl
4. Restart OS and open Ubuntu WSL
5. I get the following error
```
...
Wsl/Service/CreateInstance/ERROR_FILE_NOT_FOUND
```

### Expected Behavior

I want to see running Ubuntu WSL

### Actual Behavior

```
Wsl/Service/CreateInstance/ERROR_FILE_NOT_FOUND
```

### Diagnostic Logs

None

**Comment by asgharmughal** [comment]
Try:
wsl --unregister Ubuntu-22.04

**Reply by miniminilabcom** [reply]
> Try: wsl --unregister Ubuntu-22.04

Thanks, you saved my time. :)

**Reply by boldofyoutoassume** [reply]
Not all heroes wear capes 
![PepeCryPepeGIF](https://user-images.githubusercontent.com/98372552/188988766-3b40937f-cced-4171-933e-1987f7bd67e5.gif)

**Reply by csthenry** [reply]
you're a monster, thank you for you beautiful or unbelievable answer!!!

**Reply by reoLantern** [reply]
> Try: wsl --unregister Ubuntu-22.04

hero!

**Reply by monax90210** [reply]
> Try: wsl --unregister Ubuntu-22.04

Thank you so much for your answer!!You're the Best!
Also helped with kali-linux.

**Reply by xw-wandy** [reply]
Ohhh thanks man, you are a savior.

**Reply by aHyperGG** [reply]
Worked for me! Just had to remove the version number, thank you so much!

**Reply by MaherAbdelkhalekPTB** [reply]
I have a 2 years old project work there, will the unregistring delete the Unbunty distrubution?

**Reply by PolarBean** [reply]
This appears to delete all data from wsl. Something which should be mentioned.

**Reply by realyukii** [reply]
Anyway to avoid deletion of data? I encounter the same error right after I rename my local username

EDIT:
nvm, I just need to backup the old vdisk and override the new one
it seems the partition layout is same, so it didn't cause any corrupt/conflict

**Reply by Simkiw** [reply]
How is it not mentioned that it would result in the loss of your data...
Anyone with a solution to recover data after the distro has been unregistered ?

**Reply by airtonix** [reply]
> I have a 2 years old project work there, will the unregistring delete the Unbunty distrubution?

yes

**Comment by pmfs1** [comment]
> Try: wsl --unregister Ubuntu-22.04

Using: *wsl --unregister Ubuntu* fixed my problem. It runs perfectly now.
Thank you very much for the help.

**Reply by svenskanda** [reply]
wsl --unregister Debian the same

**Reply by Biswa96** [reply]
Try to uninstall the WSL Preview completely and then re-install from Windows Store.

**Reply by LeonLespri** [reply]
Thanks!👍

**Reply by airtonix** [reply]
> > Try: wsl --unregister Ubuntu-22.04
> 
> Using: _wsl --unregister Ubuntu_ fixed my problem. It runs perfectly now. Thank you very much for the help.

This is the equivalent of removing your lock file to solve npm install problems...

Not at all the correct solution.

**Comment by Smallthing** [comment]
me too.can this problem have a backend fix , I saw you make a beautiful Setup UI for UbuntuWSL
but many people have this issues. they even can't start the setup

**Comment by shubbu002** [comment]
It worked wonder Thanks!

**Comment by iwanglei1** [comment]
Thanks!

**Comment by burak-yldrm** [comment]
> ### Version
> Microsoft Windows [Version 10.0.22622.450]
> 
> ### WSL Version
> * [x]  WSL 2
> * [x]  WSL 1
> 
> ### Kernel Version
> Ubuntu
> 
> ### Distro Version
> Ubuntu
> 
> ### Other Software
> None
> 
> ### Repro Steps
> 1. Install wsl default distro
> 2. Uninstall wsl
> 3. Reinstall wsl
> 4. Restart OS and open Ubuntu WSL
> 5. I get the following error
> 
> ```
> ...
> Wsl/Service/CreateInstance/ERROR_FILE_NOT_FOUND
> ```
> 
> ### Expected Behavior
> I want to see running Ubuntu WSL
> 
> ### Actual Behavior
> ```
> Wsl/Service/CreateInstance/ERROR_FILE_NOT_FOUND
> ```
> 
> ### Diagnostic Logs
> None

It's best to:
After typing "wsl --list --all", find out the version (or name) of the wsl you are using.

Then you will have to type "wsl --unregister (version)"

Write the version (or name) without parentheses!

**Comment by 0xchoudhary** [comment]
It's Works 👍

**Comment by Muram1602** [comment]
Hello all, 

The command wsl --unregister (version)

Will it remove all the old data ?

In my case, I reinstalled the opensusu tumbleweed software from Microsoft store and I got the same error.
Its a 3 month project data and we don't want to loose it.

We need to mount the old files safely and the opensusu software work fine.

Is there any way to take backup before following unregister command ?

Thanks,
Muthu Raman

**Reply by airtonix** [reply]
yes

**Reply by solidguy01** [reply]
```wsl --update --pre-release ```
did fix it without loss of data for me

**Comment by mohitthakur9901** [comment]
I am really happy that I found this suggestion and My problem is solved now  I was stuck in this problem from morning

**Comment by tettolizer** [comment]
wsl --unregister ubuntu-22.04 worked for me

**Comment by busrasi** [comment]
Hello everybody,

Today, I got the same error, when I installed Docker Desktop. I just updated my pc to Windows 11. 

For Solution:

I installed Ubuntu from Windows Store. 
I restarted computer.
I executed WSL from applications like image, below:
![wsl_solution](https://github.com/microsoft/WSL/assets/23638977/6d30fb69-f37e-4a2c-bb2f-756d6f8cd40b)

Hope it works for you too!

Best Regards,
Busra Sirin

**Comment by Mahdi-Shad** [comment]
I did one more step after unregistering and installing the wsl. The last step fixed this issue ` Failed to attach disk '<HOME>\AppData\Local\Docker\wsl\main\ext4.vhdx' to WSL2: The system cannot find the file specified.`
The steps that I did:
1. wsl --unregister ubuntu
2. wsl --install
3. Restart machine
4. **Run Docker Desktop --> click troubleshoot on the top right corner--> click Clear/Purge data**
5. Restart machine
